### DISCUSSION TITLE
How to set unit_id for ModbusTCPClient

### DISCUSSION THREAD

**Discussion by maba2402** [pymodbus-dev/pymodbus#1652] [Q&A] [answered]
Hi,
I wanted to query a holding register for a ModbusTCPServer having a device ID of 2.
Here is the code:

```
from pymodbus.client import ModbusTcpClient

client = ModbusTcpClient(host='192.168.178.149',
                         port=502,
                         unit_id=2,
                         debug=True)
client.connect()

response = client.read_holding_registers(40001,5)
print(response)
client.close()
```

This results in the following print out:
`Exception Response(131, 3, IllegalAddress)`

Using wireshark gives the following details:
<img width="748" alt="image" src="https://github.com/pymodbus-dev/pymodbus/assets/6125592/ad1a9b99-de8d-455d-9d54-6e596be363c8">

Here I see Unit Identifier = 0. I was expecting 2. 
Any ideas?

**Comment by janiversen** [comment]
yes read the documentation, you are not using the proper calls.

**Reply by maba2402** [reply, marked as answer]
Hi,
I found the issue. It was a problem with so called alias adresses for the holding registers.
Subtracting 40001 from the register adresse solved the problem.
Sorry for bothering you.

Setting the device identifier was not the issue. This works BTW with

>  rr = client.read_holding_registers(1633, 1, slave=2)

BR
Marc
